### DISCUSSION TITLE
Better ColorScheme Defaults?

### DISCUSSION THREAD

**Discussion by elibroftw** [mantinedev/mantine#1125] [Feedback]
https://mantine.dev/theming/dark-theme/#colorschemeprovider

Not sure if you noticed, but shouldn't this be the default? Why are there two providers and for each you have to provide the colorScheme? It seems very counter-productive. I think something should be done to simplify it. My boilerplate code is quickly expanding!

Even after using this colorScheme provider, createStyles' theme has the wrong value set for colorScheme. How do I get createStyles to return the correct theme.colorScheme?

**Comment by rtivital** [comment]
If createStyles does not return correct color scheme – you've setup something incorrectly, you can find full example in template – https://github.com/mantinedev/mantine-next-template

**Reply by elibroftw** [reply]
I had looked at the template already but it doesn't use create styles.

Is this like the notification provider where the themes color scheme doesn't update for the current component but only for the children?

If so can you recommend people to put these providers in main.jsx rather than app.jsx?

**Reply by rtivital** [reply]
I do not know what your issue is, you did not provide any examples.

**Reply by elibroftw** [reply]
https://github.com/elibroftw/modern-desktop-app-template/blob/master/src/App.jsx

**Reply by rtivital** [reply]
1. Do not use createStyles within components, see https://ui.mantine.dev/ if you need an example of correct usage
2. createStyles depend on MantineProvider context. You are calling it ouside of the context and thus you will always get default theme value instead of one defined on MantineProvider.

**Reply by elibroftw** [reply]
1. It's preferred though
2. This is why I said maybe your docs should recommend the provider to be in the entry point. No? As it stands, even your template has the providers in app.tsx. I don't know about you, but app.jsx should not be used as a second setup file if index/main already exists.

**Reply by elibroftw** [reply]
And what should I be using other than create styles within components?

Inline styling?

**Reply by rtivital** [reply]
If you need to use extra logic in _app.tsx wrap this logic with component and then use it in _app.tsx:

```tsx
function App() {
  return (
    <MantineProvider>
      <ExtraLogic>
        {children}
       </ExtraLogic>
    </MantineProvider>
  );
}
```

**Reply by elibroftw** [reply]
How will that fix the problem? What would even go into extra logic? My entire app shell? The only real child is an app shell 😂!

**Reply by rtivital** [reply]
At least createStyles that will get MantineProvider context.

**Reply by elibroftw** [reply]
But the styles need to be applied to app shell. So then my solution is better to just use the providers in the entry point to avoid hiccups. Modules like i18n tell users to edit their entry script. 

You should strongly consider telling all users to put the providers in the entry rather than the app. This will save people time as well as reduce redundancy.

**Reply by elibroftw** [reply]
Also I don't know if anyone has mentioned it to you @rtivital, but every time I go to https://mantine.dev/ or refresh the page, the light theme shows for 1 second before the dark theme "turns on." That's why I didn't ask how you do this colorScheme management with your websites. My method doesn't work because I'd have to know the value of colorScheme in `main.jsx` which isn't supposed to imo have any logic.

**Reply by elibroftw** [reply]
My other question is if there is a `getTheme('dark')` function or `mantine.useColorSchem(dark')`?

**Comment by elibroftw** [comment]
I was forced to do this :(
```js
// top level
function getClasses(colorScheme) {
  const { classes } = createStyles(theme => ({
    // ...
  }))();
  return classes;
}
```
